server/src/internalClusterTest/java/org/elasticsearch/snapshots/GetSnapshotsIT.java

```diff
@@ -745,4 +762,196 @@ private static GetSnapshotsRequestBuilder baseGetSnapshotsRequest(String[] repoN
+            getSnapshotsRequest.offset(between(0, selectedSnapshots.size() + 1));
+        }
+
+        // ?size parameter
+        if (randomBoolean()) {
+            getSnapshotsRequest.size(between(1, selectedSnapshots.size() + 1));
+        }
+
+        // compute the expected offset and size of the returned snapshots as indices in selectedSnapshots:
+        final var expectedOffset = Math.min(selectedSnapshots.size(), skippedByFromSortValue + getSnapshotsRequest.offset());
+        final var expectedSize = Math.min(
+            selectedSnapshots.size() - expectedOffset,
+            getSnapshotsRequest.size() == GetSnapshotsRequest.NO_LIMIT ? Integer.MAX_VALUE : getSnapshotsRequest.size()
+        );
+
+        // get the actual response
+        final GetSnapshotsResponse getSnapshotsResponse = safeAwait(
+            l -> client().execute(TransportGetSnapshotsAction.TYPE, getSnapshotsRequest, l)
+        );
+
+        // verify it returns the expected results
+        assertEquals(
+            selectedSnapshots.stream().skip(expectedOffset).limit(expectedSize).map(SnapshotInfo::snapshotId).toList(),
+            getSnapshotsResponse.getSnapshots().stream().map(SnapshotInfo::snapshotId).toList()
+        );
+        assertEquals(expectedSize, getSnapshotsResponse.getSnapshots().size());
+        assertEquals(selectedSnapshots.size() - skippedByFromSortValue, getSnapshotsResponse.totalCount());
+        assertEquals(selectedSnapshots.size() - expectedOffset - expectedSize, getSnapshotsResponse.remaining());
+        assertEquals(getSnapshotsResponse.remaining() > 0, getSnapshotsResponse.next() != null);
+
+        // now use ?after to page through the rest of the results
+        var nextRequestAfter = getSnapshotsResponse.next();
+        var nextExpectedOffset = expectedOffset + expectedSize;
+        var remaining = getSnapshotsResponse.remaining();
+        while (nextRequestAfter != null) {
+            final var nextSize = between(1, remaining);
+            final var nextRequest = new GetSnapshotsRequest(TEST_REQUEST_TIMEOUT, requestedRepositories, requestedSnapshots)
+                // same name filters, same ?sort and ?order params, new ?size, but no ?offset or ?from_sort_value because of ?after
+                .sort(sortKey)
+                .order(order)
+                .size(nextSize)
+                .after(SnapshotSortKey.decodeAfterQueryParam(nextRequestAfter));
+            final GetSnapshotsResponse nextResponse = safeAwait(l -> client().execute(TransportGetSnapshotsAction.TYPE, nextRequest, l));
+
+            assertEquals(
+                selectedSnapshots.stream().skip(nextExpectedOffset).limit(nextSize).map(SnapshotInfo::snapshotId).toList(),
+                nextResponse.getSnapshots().stream().map(SnapshotInfo::snapshotId).toList()
+            );
+            assertEquals(nextSize, nextResponse.getSnapshots().size());
+            assertEquals(selectedSnapshots.size(), nextResponse.totalCount());
```

Comment:
This is not 
```suggestion
            assertEquals(selectedSnapshots.size() - skippedByFromSortValue, nextResponse.totalCount());
```
?

Reply:
Correct. We're not even specifying `?from_sort_value` in this request, so we're seeing every matching snapshot in the total. I think we probably should not subtract these skipped items from the original total either, it seems fairly inconsistent to do what we do today, but this behaviour dates back quite some time so we'd need to discuss any changes. It's noted in this comment:

https://github.com/elastic/elasticsearch/blob/58ff48ce1f95df37fb27efc0587d3c6663650807/server/src/internalClusterTest/java/org/elasticsearch/snapshots/GetSnapshotsIT.java#L866-L867